### ISSUE TITLE
Implement "discard"

### ISSUE THREAD

**Issue by mausch** [fsharp/FsCheck#88]
QuickCheck has (since 2.7) a data type Discard that forces the test case to be discarded when the property returns it. This is useful because as the hackage doc says "[Normally you should use ==>, but if for some reason this isn't possible (e.g. you are deep inside a generator), use discard instead.](http://hackage.haskell.org/package/QuickCheck-2.7.6/docs/Test-QuickCheck.html#v:discard)".

QuickCheck has both a `Testable Discard` instance and a `discard` function that throws an exception. 

Now, FsCheck already has a Testable instance for `Result`, which seems like it would cover this in principle, but returning a Result is atypical and seems rather cumbersome unless there are some simplified constructors that I'm missing.

Also Testables are restricted in C# to unit and boolean AFAICT, but I guess that should be addressed in https://github.com/fsharp/FsCheck/issues/63 .

In summary: what do you think would be the best way to implement this?

**Comment by kurtschelfthout** [comment]
Do we need a Discard instance, unclear how it would be used? Seems like having a discard function that throws DiscardException that is caught and handled specially would be more idiomatic.

**Comment by mausch** [comment]
Yes, throwing the exception would work too of course. I just thought it would complicate the implementation since it makes this a special case whereas returning a `Result` (using some simplified constructor for this purpose though I have no idea about everything that goes in there) is already supported.

Anyway, DiscardException it is then. Now I need to find the time to actually implement this :-)

**Comment by kurtschelfthout** [comment]
I see. Yes, maybe you're right. Have a look at the Testable module; may be as easy as calling Res.rejected (there is an internal module Res that allows you to create Result instances; rejected means discarded).

**Comment by mausch** [comment]
Ah, yes, those are the simplified constructors I was thinking about. Does it not matter that Stamp, Labels and Arguments are all empty? Also, if we expose `Res.rejected` (or an alias named `discard`), it seems we should also expose the other `Result` constructors so that the user can at least say that a test case is ok to check. From that point of view, the exception seems simpler, right?

**Comment by kurtschelfthout** [comment]
Yes, agreed. That why I was wondering how to use a Discard type. 

The stamps etc would be filled in by other combinators, that part should work.

**Comment by mausch** [comment]
I ended up writing the test/generator in a different way so I don't need it any more, at least for now. Still, I think it might come back again so let's leave this issue open...

**Comment by kurtschelfthout** [comment]
This is in 2.0.0-alpha.
